src/openloco/companymgr.cpp

```diff
@@ -103,4 +109,78 @@ namespace openloco::companymgr
             }
         }
     }
+
+    // 0x004383ED
+    void updateOwnerStatus()
+    {
+        if (openloco::is_title_mode() || openloco::is_editor_mode())
+        {
+            return;
+        }
+
+        auto company = companymgr::get(_updating_company_id);
+        if (company == nullptr)
+        {
+            return;
+        }
+
+        company->var_12 += 1;
+        if ((company->var_12 & 0x7F) != 0)
+            return;
+
+        for (size_t i = 0; i < WindowManager::count(); i++)
+        {
+            auto w = WindowManager::get(i);
+
+            if (w->type != WindowType::vehicle)
+                continue;
+
+            auto vehicle = thingmgr::get<openloco::vehicle>(w->number);
+            if (vehicle->x == -0x8000)
+                continue;
+
+            if (vehicle->owner != _updating_company_id)
+                continue;
+
+            game_commands::do_73(vehicle->id);
+            return;
+        }
+
+        auto main = WindowManager::getMainWindow();
+        if (main == nullptr)
+            return;
+
+        auto viewport = main->viewports[0];
+        if (viewport == nullptr)
+            return;
+
+        int16_t posX = viewport->x + viewport->width / 2;
+        int16_t posY = viewport->y + viewport->height / 2;
+
+        registers r1;
+        r1.ax = posX;
+        r1.bx = posY;
+        call(0x0045F1A7, r1);
+        ui::viewport* vp = (ui::viewport*)r1.edi;
+
+        if (posX != -0x8000 && viewport == vp)
+        {
+            // The result of this function appears to be ignored
+            map::tile_element_height(posX, posY);
```

Comment:
I think you're right, calling `tile_element_height` seems to be a lot of wasted cycles — especially the bit after the jump into `off_4FD250`.

While `ecx` is reset directly after the call, `tile_element_height` does `cmp bl, 0` (`0x45F1B3`) straight after calling `get_map_coordinates_from_pos`, and if zero sets `add esp, 8` and `mov ax, 8000h` (`0x45F992`).

At `0x43849D` this `ax` is compared to again, so it appears the call is useful to that end, at least.

Reply:
![image](https://user-images.githubusercontent.com/737603/56425554-1dcbaa00-62b5-11e9-94f1-d89a8c98a3d1.png)

I think you're talking about a different function, namelijk the top `screen_get_map_xy`. I'm talking about the yellow-marked one. The output of this function is the Z-height of the surface at the passed (3d) X/Y coordinates, but it's never used by the game command. This part of the function seems very similar to `window_rotate_camera` in OpenRCT2.